helpers.lua

```diff
@@ -299,6 +299,46 @@ function log_access(user, uri)
   end
 end
 
+-- Check if the request contains the admin token
+function is_admin()
+
+  local admin_token_header = ngx.req.get_headers()["SSOwat-Admin-Token"]
+
+  if admin_token_header == nil then
+      return false
+  end
+
+  -- FIXME : ideally we should also be checking that the permissions for this file are something like 600 + appropriate owner...
+  local admin_token_file = io.open(admin_token_path, "r")
+  if admin_token_file == nil then
+      -- N.B. these messages should use the same syntax as in authenticate()
+      ngx.log(ngx.ERR, "Connection failed for: admin")
+      logger.error("Authentication failure for user admin from "..ngx.var.remote_addr)
+      return false
+  end
+  local token = admin_token_file:read("*all")
```

Comment:
What about timing attack here ?

Reply:
What do you mean, how do you time-attack a file reading ?